src/hotspot/share/prims/forte.cpp

```diff
@@ -563,22 +564,26 @@ extern "C" {
 JNIEXPORT
 void AsyncGetCallTrace(ASGCT_CallTrace *trace, jint depth, void* ucontext) {
 
-  JavaThread* thread;
+  // Can't use thread_from_jni_environment as it may also perform a VM exit check that is unsafe to
+  // do from this context.
+  Thread* raw_thread = Thread::current_or_null_safe();
 
-  if (trace->env_id == NULL ||
-      (thread = JavaThread::thread_from_jni_environment(trace->env_id))->is_exiting()) {
+  if (trace->env_id == NULL || raw_thread == NULL || !raw_thread->is_Java_thread() || ((JavaThread*)raw_thread)->is_exiting()) {
```

Comment:
use `rawThread->as_JavaThread()` not a plain cast.

Reply:
this method does not exist

Reply:
Sorry I meant: `JavaThread::cast(rawThread)`

Reply:
You missed changing the cast here. You could also save the result for use below.

Reply:
You're right, I missed that cast.